src/platform/silabs/tracing/SilabsTracing.cpp

```diff
@@ -0,0 +1,277 @@
+{
+    size_t index     = static_cast<size_t>(aOperation);
+    auto & watermark = mWatermarks[index];
+
+    VerifyOrReturnError(isLogInitialized(), CHIP_ERROR_UNINITIALIZED);
+    ChipLogProgress(DeviceLayer,
+                    "Trace %zu: TotalCount=%" PRIu32 ", SuccessFullCount=%" PRIu32 ", MaxTime=%" PRIu32 ", MinTime=%" PRIu32
+                    ", AvgTime=%" PRIu32 ", CountAboveAvg=%" PRIu32 "",
+                    index, watermark.mTotalCount, watermark.mSuccessfullCount, watermark.mMaxTimeMs.count(),
+                    watermark.mMinTimeMs.count(), watermark.mMovingAverage.count(), watermark.mCountAboveAvg);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SilabsTracer::OutputAllWaterMarks()
+{
+    for (size_t i = 0; i < kNumTraces; ++i)
+    {
+        CHIP_ERROR err = OutputWaterMark(static_cast<TimeTraceOperation>(i));
+        if (err != CHIP_NO_ERROR)
+        {
+            return err;
+        }
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SilabsTracer::TraceBufferFlushAll()
+{
+    auto * current = mTimeTrackerList.head;
+    while (current != nullptr)
+    {
+        // We do not want to loose the traces if the logs are not initialized
+        ReturnErrorOnFailure(OutputTimeTracker(current->mValue));
+        current = current->mpNext;
+    }
+
+    TraceBufferClear();
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SilabsTracer::TraceBufferFlushByOperation(TimeTraceOperation aOperation)
+{
+
+    auto * current = mTimeTrackerList.head;
+    auto * prev    = static_cast<chip::SingleLinkedListNode<TimeTracker> *>(nullptr);
+    while (current != nullptr)
+    {
+        if (current->mValue.mOperation == aOperation)
+        {
+            ReturnErrorOnFailure(OutputTimeTracker(current->mValue));
+            if (prev == nullptr)
+            {
+                mTimeTrackerList.head = current->mpNext;
+            }
+            else
+            {
+                prev->mpNext = current->mpNext;
+            }
+            auto * temp = current;
+            current     = current->mpNext;
+            free(temp);
+            mBufferedTrackerCount--;
+        }
+        else
+        {
+            prev    = current;
+            current = current->mpNext;
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+// Save the traces in the NVM
+CHIP_ERROR SilabsTracer::SaveWatermarks()
+{
+    VerifyOrReturnError(nullptr != mStorage, CHIP_ERROR_INCORRECT_STATE);
+    // TODO implement the save of the watermarks in the NVM
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SilabsTracer::LoadWatermarks()
+{
+    VerifyOrReturnError(nullptr != mStorage, CHIP_ERROR_INCORRECT_STATE);
+    // TODO implement the load of the watermarks from the NVM
+    return CHIP_NO_ERROR;
+}
+
+#if CONFIG_BUILD_FOR_HOST_UNIT_TEST
+CHIP_ERROR SilabsTracer::GetTraceCount(size_t & count) const
+{
+    count = mBufferedTrackerCount;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SilabsTracer::GetTraceByIndex(size_t index, const char *& trace) const
```

Comment:
This should be a method of TimeTrackerList.

Reply:
Yes, but this method got removed in my ongoing branch : https://github.com/SiliconLabsSoftware/matter_sdk/tree/feature/silabs_tracing_logs_formating 

So to reduce the conflict from the rebase, I will keep it as is here.